src/main/tv/codely/scala_intro_examples/lesson_09_oop/exercise_solutions/joan_miralles_solution/Email.scala

```diff
@@ -0,0 +1,23 @@
+package tv.codely.scala_intro_examples.lesson_09_oop.exercise_solutions.joan_miralles_solution
+
+import java.security.InvalidParameterException
+
+object Email {
+  def apply(completeEmail: String): Email = {
+    val atIndex = completeEmail.indexOf("@")
+    if (completeEmail.isEmpty || !completeEmail.contains("@") || !completeEmail.substring(completeEmail.indexOf("@")).contains(".")) {
+      throw new InvalidParameterException(s"$completeEmail is not a valid e-mail")
+    }
+    Email(completeEmail.substring(0, atIndex), completeEmail.substring(atIndex + 1, completeEmail.length))
+  }
+}
+
+final case class Email(local: String = "soporte", domain: String = "codely.tv") {
+
+  private val email = s"$local@$domain"
+
+  def getEmail: String = {
```

Comment:
Unnecessary getter. It's not idiomatic in Scala to define getters and setter methods since our case classes are immutable by definition 🙂

In any case, it wouldn't have to be defined as `def` since its value will not change over time. So based on that, we don't need to evaluate this expression on each `getEmail` call, making it possible to be a `val getEmail`.